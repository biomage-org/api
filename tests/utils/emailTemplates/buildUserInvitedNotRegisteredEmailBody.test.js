const { ACCOUNT_ID } = require('../../../src/api.v2/constants');
const config = require('../../../src/config');
const buildUserInvitedNotRegisteredEmailBody = require('../../../src/utils/emailTemplates/buildUserInvitedNotRegisteredEmailBody');

const testUserEmail = 'test@email.com';
const mockInviterUser = { email: 'inviter@example.com' };

process.env.DOMAIN_NAME = 'localhost.test';

describe('buildUserInvitedNotRegisteredEmailBody', () => {
  it('Should build the correct message', () => {
    const emailParams = buildUserInvitedNotRegisteredEmailBody(testUserEmail, mockInviterUser);
    expect(emailParams).toMatchSnapshot();
  });

  it('Should build the correct message for HMS', () => {
    config.awsAccountId = ACCOUNT_ID.HMS;

    const emailParams = buildUserInvitedNotRegisteredEmailBody(testUserEmail, mockInviterUser);
    expect(emailParams).toMatchSnapshot();
  });
});
