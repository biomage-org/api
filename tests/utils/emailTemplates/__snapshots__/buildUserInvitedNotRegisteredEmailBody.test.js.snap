// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildUserInvitedNotRegisteredEmailBody Should build the correct message 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "test@email.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <p>
              Hello, <br/>
              Your collaborator inviter@example.com has invited you to explore their single cell RNA-seq project in Cellenics. <br/><br/>
              You need to create a Cellenics account in order to access the project. <br/>
              Register at <a href=\\"https://test.com\\">https://test.com</a> using this email address.<br/> <br/>

              Cellenics is a user-friendly online tool for single cell RNA-seq data analysis. <br/>
              The platform is designed specifically for biologists, and it's completely free for academic researchers.<br/><br/>
              More information about Cellenics can be found at <a href=\\"https://biomage.net\\">biomage.net</a>.<br/><br/>
              If you need help or have any questions, please contact us at hello@biomage.net. <br/><br/>
              Best Regards, <br/>
              Cellenics team
            <p/>
        </body>
      </html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <p>
              Hello, <br/>
              Your collaborator inviter@example.com has invited you to explore their single cell RNA-seq project in Cellenics. <br/><br/>
              You need to create a Cellenics account in order to access the project. <br/>
              Register at <a href=\\"https://test.com\\">https://test.com</a> using this email address.<br/> <br/>

              Cellenics is a user-friendly online tool for single cell RNA-seq data analysis. <br/>
              The platform is designed specifically for biologists, and it's completely free for academic researchers.<br/><br/>
              More information about Cellenics can be found at <a href=\\"https://biomage.net\\">biomage.net</a>.<br/><br/>
              If you need help or have any questions, please contact us at hello@biomage.net. <br/><br/>
              Best Regards, <br/>
              Cellenics team
            <p/>
        </body>
      </html>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Invitation to join a project in Cellenics",
    },
  },
  "Source": "notification@localhost.test",
}
`;

exports[`buildUserInvitedNotRegisteredEmailBody Should build the correct message for HMS 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "test@email.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <p>
              Hello, <br/>
              Your collaborator inviter@example.com has invited you to explore their single cell RNA-seq project in Cellenics. <br/><br/>
              You need to create a Cellenics account in order to access the project. <br/>
              Register at <a href=\\"https://test.com\\">https://test.com</a> using this email address.<br/> <br/>

              Cellenics is a user-friendly online tool for single cell RNA-seq data analysis. <br/>
              The platform is designed specifically for biologists, and it's completely free for academic researchers.<br/><br/>
              Best Regards, <br/>
              Cellenics team
            <p/>
        </body>
      </html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <p>
              Hello, <br/>
              Your collaborator inviter@example.com has invited you to explore their single cell RNA-seq project in Cellenics. <br/><br/>
              You need to create a Cellenics account in order to access the project. <br/>
              Register at <a href=\\"https://test.com\\">https://test.com</a> using this email address.<br/> <br/>

              Cellenics is a user-friendly online tool for single cell RNA-seq data analysis. <br/>
              The platform is designed specifically for biologists, and it's completely free for academic researchers.<br/><br/>
              Best Regards, <br/>
              Cellenics team
            <p/>
        </body>
      </html>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Invitation to join a project in Cellenics",
    },
  },
  "Source": "alex_pickering@hms.harvard.edu",
}
`;
